testing/integration/container_cmd_test.go

```diff
@@ -258,3 +275,175 @@ func TestContainerCMDWithAVeryLongStatePath(t *testing.T) {
 		})
 	}
 }
+
+func TestContainerCMDEventToStderr(t *testing.T) {
+	info := define.Require(t, define.Requirements{
+		Stack: &define.Stack{},
+		Local: false,
+		Sudo:  true,
+		OS: []define.OS{
+			{Type: define.Linux},
+		},
+		Group: "container",
+	})
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
+	defer cancel()
+
+	agentFixture, err := define.NewFixtureFromLocalBuild(t, define.Version())
+	require.NoError(t, err)
+
+	_, outputID := createMockESOutput(t, info)
+	policyID, enrollmentAPIKey := createPolicy(t, ctx, agentFixture, info, outputID)
+
+	fleetURL, err := fleettools.DefaultURL(ctx, info.KibanaClient)
+	if err != nil {
+		t.Fatalf("could not get Fleet URL: %s", err)
+	}
+
+	env := []string{
+		"FLEET_ENROLL=1",
+		"FLEET_URL=" + fleetURL,
+		"FLEET_ENROLLMENT_TOKEN=" + enrollmentAPIKey,
+		"STATE_PATH=" + agentFixture.WorkDir(),
+		// That is what we're interested in testing
+		"EVENTS_TO_STDERR=true",
+	}
+
+	cmd := prepareContainerCMD(t, ctx, agentFixture, info, env)
+	addLogIntegration(t, info, policyID, "/tmp/flog.log")
+	generateLogFile(t, "/tmp/flog.log", time.Second/2, 100)
+
+	t.Logf(">> running binary with: %v", cmd.Args)
+	if err := cmd.Start(); err != nil {
+		t.Fatalf("error running container cmd: %s", err)
+	}
+
+	agentOutput := cmd.Stderr.(*strings.Builder)
+
+	assert.Eventuallyf(t, func() bool {
+		// This will return errors until it connects to the agent,
+		// they're mostly noise because until the agent starts running
+		// we will get connection errors. If the test fails
+		// the agent logs will be present in the error message
+		// which should help to explain why the agent was not
+		// healthy.
+		err := agentFixture.IsHealthy(ctx)
+		return err == nil
+	},
+		2*time.Minute, time.Second,
+		"Elastic-Agent did not report healthy. Agent status error: \"%v\", Agent logs\n%s",
+		err, agentOutput,
+	)
+
+	assert.Eventually(t, func() bool {
+		agentOutputStr := agentOutput.String()
+		scanner := bufio.NewScanner(strings.NewReader(agentOutputStr))
+		for scanner.Scan() {
+			if strings.Contains(scanner.Text(), "Cannot index event publisher.Event") {
```

Comment:
This doesn't actually test the logic that requires the agent re-exec because the settings don't change.

Can you modify it to test that as well? That could potentially be tested without the container command since I don't think the re-exec is container specific.

Reply:
> This doesn't actually test the logic that requires the agent re-exec because the settings don't change.

No it does not. The intent of the test is to test the environment variable `EVENTS_TO_STDERR`

I thought I had added a test in the handler for that, but it seems I didn't :thinking: . I'll add another test to test the re-exec path.